map_model/src/pathfind/walking.rs

```diff
@@ -299,21 +299,25 @@ fn make_input_graph(
             let src = map.get_l(t.id.src);
             let dst = map.get_l(t.id.dst);
             let from =
-                WalkingNode::SidewalkEndpoint(src.get_directed_parent(), src.dst_i == t.id.parent);
+                nodes.get(WalkingNode::SidewalkEndpoint(src.get_directed_parent(), src.dst_i == t.id.parent));
             let to =
-                WalkingNode::SidewalkEndpoint(dst.get_directed_parent(), dst.dst_i == t.id.parent);
-            let mut cost = t.geom.length()
-                / PathStep::Turn(t.id).max_speed_along(max_speed, PathConstraints::Pedestrian, map);
-            if t.turn_type == TurnType::UnmarkedCrossing {
-                // TODO Add to RoutingParams
-                cost = 3.0 * cost;
+                nodes.get(WalkingNode::SidewalkEndpoint(dst.get_directed_parent(), dst.dst_i == t.id.parent));
+            for (step, pair) in [
+                (PathStep::Turn(t.id), (from, to)),
+                (PathStep::ContraflowTurn(t.id), (to, from)),
+            ] {
+                let mut cost = t.geom.length()
```

Comment:
Nit: I'd leave this outside the loop, just to visually make it more clear that it's the same cost for both directions of the turn

Reply:
After rereading I eliminated the loop and replaced it with two calls to add_edge